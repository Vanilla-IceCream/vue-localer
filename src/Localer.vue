<script lang="ts" setup>
import { computed, useSlots, h } from 'vue';

const props = defineProps<{ message: string }>();

const slots = useSlots();

const slotsName = computed(() => {
  return props.message?.match(/{([^}]+)?}/gm)?.map((s) => s.slice(1, s.length - 1));
});

const VNode = () => {
  const _slotsName = slotsName.value || [];

  const el = props.message?.split(/{([^}]+)?}/gm);

  const result = el.map((item) => {
    if (_slotsName.includes(item)) {
      console.log(item);

      return h('span', slots[item]?.());
    }

    return h('span', null, item);
  });

  return result;
};
</script>

<template>
  <VNode />
</template>
