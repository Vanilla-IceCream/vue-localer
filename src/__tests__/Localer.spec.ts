import { test, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import Localer from '../Localer.vue';

test('Localer - en', () => {
  const en = {
    term: `I have read and agree to the {LA}.`,
    LA: 'License Agreement',
  };

  const wrapper = mount(Localer, {
    props: {
      message: en.term,
    },
    slots: {
      LA: `<a href="https://...">${en.LA}</a>`,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});

test('Localer - ja', () => {
  const ja = {
    term: `{LA}を読み、同意します。`,
    LA: 'ライセンス契約',
  };

  const wrapper = mount(Localer, {
    props: {
      message: ja.term,
    },
    slots: {
      LA: `<a href="https://...">${ja.LA}</a>`,
    },
  });

  expect(wrapper.html()).toMatchSnapshot();
});
