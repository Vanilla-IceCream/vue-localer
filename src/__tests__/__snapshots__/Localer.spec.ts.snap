// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Localer - en 1`] = `
"<span>I have read and agree to the </span>
<span><a href="https://...">License Agreement</a></span>
<span>.</span>"
`;

exports[`Localer - ja 1`] = `
"<span></span>
<span><a href="https://...">ライセンス契約</a></span>
<span>を読み、同意します。</span>"
`;
